ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/Send.hs

```diff
@@ -218,6 +254,98 @@ createAccountSelector uacis = do
 
   return AccountSelector{..}
 
+createNewReciever :: UiLangFace -> Send -> IO ()
+createNewReciever uiLangFace@UiLangFace{..} s@Send{..} = do
+  recieverWidget <- QWidget.new
+  layout <- createLayout recieverWidget
+  QLayout.setContentsMarginsRaw layout 0 18 0 18
+
+  amountEdit <- QLineEdit.new
+  unitLabel <- QLabel.newWithText ("ADA" :: String)
+  QObject.setObjectName unitLabel ("amountLabel" :: String)
+  QObject.setObjectName amountEdit ("sendAmountEdit" :: String)
+  QLineEdit.setPlaceholderText amountEdit ("0" :: String)
+
+  let UiWalletFace{..} = uiWalletFace
+
+  cLocale <- QLocale.newWithName ("C" :: String)
+  QLocale.setNumberOptions cLocale QLocale.RejectGroupSeparator
+  validator <- QDoubleValidator.new
+  QValidator.setLocale validator cLocale
+  QDoubleValidator.setBottom validator 0
+  QDoubleValidator.setDecimals validator uiCoinPrecision
+  QLineEdit.setValidator amountEdit validator
+
+  amountLayout <- QHBoxLayout.new
+  QBoxLayout.addWidget amountLayout amountEdit
+  QBoxLayout.addWidget amountLayout unitLabel
+  removeButton <- QPushButton.new
+  void $ setProperty removeButton ("styleRole" :: Text) ("removeButton" :: Text)
+  QBoxLayout.addWidget amountLayout removeButton
+  QBoxLayout.setSpacing amountLayout 2
+  addressEdit <- QLineEdit.new
+  QLineEdit.setPlaceholderText addressEdit ("Type or paste receiver address here" :: String)
+
+  addRowLayout layout addressEdit amountLayout
+  addSeparator layout
+
+  QBoxLayout.addWidget recieversLayout recieverWidget
+
+  connect_ amountEdit QLineEdit.textChangedSignal $ \_ -> do
+    revalidate s
+    recalcTotal uiLangFace s
+  connect_ addressEdit QLineEdit.textChangedSignal $ \_ -> revalidate s
+
+  let r = Reciever{..}
+
+  connect_ removeButton QAbstractButton.clickedSignal $ \_ -> do
+    removeReciever uiLangFace r s
+
+  modifyIORef recievers (r :)
+  QWidget.adjustSize recieversWidget
+  QWidget.adjustSize send
+  revalidate s
+
+removeReciever :: UiLangFace -> Reciever -> Send -> IO ()
+removeReciever uiLangFace@UiLangFace{..} Reciever{..} s@Send{..} = do
+  enabledRecievers <- getEnabledRecievers s
+  QWidget.setEnabled recieverWidget False
+  QWidget.setVisible recieverWidget False
+  revalidate s
+  recalcTotal uiLangFace s
+  -- To prevent removing last reciever
+  if length enabledRecievers == 1
+    then do
+      createNewReciever uiLangFace s
+    else do
+      QWidget.adjustSize recieversWidget
+      QWidget.adjustSize send
+
+getEnabledRecievers :: Send -> IO [Reciever]
+getEnabledRecievers Send{..} = do
+  r <- readIORef recievers
+  filterM (\Reciever{..} -> QWidget.isEnabled recieverWidget) r
+
+recalcTotal :: UiLangFace -> Send -> IO ()
+recalcTotal UiLangFace{..} s@Send{..} = do
+  enabledRecievers <- getEnabledRecievers s
+  textAmounts <- mapM (\Reciever{..} -> fromString <$> QLineEdit.text amountEdit) enabledRecievers
+  let amounts = map Unsafe.fromJust (filter isJust (map (\a -> normalize <$> readMaybe a) textAmounts))
```

Comment:
Brr, please no. Use `catMaybes`. Btw, the lambda can be just `fmap normalize . readMaybe`.